### PR TITLE
New deploy detail page

### PR DESCRIPTION
**Pull request by kwm4385**
image

Deploy detail react/redux
@tpetr @wolfd @Calvinp

### CODE REVIEW

**Comment by tpetr** [comment]
😆 💦

File: SingularityUI/app/controllers/DeployDetail.es6 [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+import { FetchAction as DeployFetchAction} from '../actions/api/deploy';
+import { fetchTask as TaskFetchAction} from '../actions/api/task';
+import { FetchForDeployAction } from '../actions/api/tasks';
+import { FetchForDeploy as TaskHistoryFetchForDeploy } from '../actions/api/taskHistory';
+
+class DeployDetailController extends Controller {
+
+    initialize({store, requestId, deployId}) {
+        app.showPageLoader()
+        this.title('Status');
```
Comment (tpetr) [inline SingularityUI/app/controllers/DeployDetail.es6]: need to update this value

File: SingularityUI/app/controllers/DeployDetail.es6 [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        });
+
+        let tasksPromise = this.store.dispatch(FetchForDeployAction.trigger(requestId, deployId));
+        tasksPromise.then(() => {
+          for (let t of store.getState().api.activeTasksForDeploy.data) {
+            this.store.dispatch(TaskFetchAction(t.taskId.id));
+          }
+        });
+
+        this.store.dispatch(TaskHistoryFetchForDeploy.trigger(requestId, deployId, 5, 1));
```
Comment (tpetr) [inline SingularityUI/app/controllers/DeployDetail.es6]: it'd be a better experience for the user if we held of calling `showView()` until all the promises (or atleast, all the ones for visible things) have been resolved. i'd suggest coalescing the individual promises into an uber-promise using `Promise.all()`. and then calling `showView()` once the uber-promise has resolved.

File: SingularityUI/app/controllers/DeployDetail.es6 [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+          this.setView(new DeployView(store));
+          app.showView(this.view);
+        });
+
+        let tasksPromise = this.store.dispatch(FetchForDeployAction.trigger(requestId, deployId));
+        tasksPromise.then(() => {
+          for (let t of store.getState().api.activeTasksForDeploy.data) {
+            this.store.dispatch(TaskFetchAction(t.taskId.id));
+          }
+        });
```
Comment (tpetr) [inline SingularityUI/app/controllers/DeployDetail.es6]: there are identical `dispatch()`es in `refresh()` -- can we call `refresh()` instead?

**Comment by tpetr** [comment]
- Active Tasks table shouldn't collapse
- Updated column of Active Tasks and Task History tables always says "a few seconds ago" which isnt always accurate
- Latest Healthchecks table is showing healthchecks for a previous web service (that part of the store needs to be be cleared out)

**Comment by Calvinp** [comment]
>Updated column of Active Tasks and Task History tables always says "a few seconds ago" which isnt always accurate

You should look outside of your code for the fix to this one - timestamps are broken elsewhere too.

**Comment by kwm4385** [comment]
^ Above all should be fixed

**Comment by tpetr** [comment]
@calvinp I disagree -- this PR should keep its focus on reactifying and reduxifying the deploy detail page. It should be up to the owner (ex. Me for racks and slaves page) to fix their own mistakes or omissions. Though, additional PR's are welcome if folks have extra bandwidth too! 😃

**Comment by kwm4385** [comment]
At least in this case the timestamp issue was simply because I had the wrong field. I'm not aware of a general problem with timestamps across the app.

**Comment by tpetr** [comment]
- Don't show Latest Healthchecks table if the Request type isnt `SERVICE`
- Task History shouldn't collapse either

**Comment by tpetr** [comment]
^ should be good to go once those last two issues are squared away, nice work!

**Comment by kwm4385** [comment]
^ also removed collapse from task history

**Comment by kwm4385** [comment]
🔨

**Pull request merged**